test/unit/library/formats.jl

```diff
@@ -0,0 +1,174 @@
+function test_format_hints()
+    @testset "⋅ Format Hints" begin
+        @test QUBOTools.format(:bool, :json) isa QUBOTools.BQPJSON
+        @test QUBOTools.format("file.bool.json") isa QUBOTools.BQPJSON
+
+        @test QUBOTools.format(:spin, :json) isa QUBOTools.BQPJSON
+        @test QUBOTools.format("file.spin.json") isa QUBOTools.BQPJSON
+
+        # @test QUBOTools.format(:hfs) isa QUBOTools.HFS
+        # @test QUBOTools.format("file.hfs") isa QUBOTools.HFS
```

Comment:
What's up with these commented tests?

Reply:
I'd like to discuss HFS and MiniZinc support. In summary:
I was thinking that HFS is very dwave specific and this format could be implemented and maintained inside DWave.jl
MiniZinc write support should continue but reading should be dropped. Its parser was ported from bqpjson and is limited to the subset of the language that's written, but subtle changes in the way mzn programs are written may break the parser. The ideal scenario would be to leverage a complete mzn engine. Unfortunately, MiniZinc.jl will also only write mzn files, not read from them.
I will open an issue on this later.

Reply:
Feel free to put this into the new issue.
I do not see the point in transforming the file inside DWave.jl. Even if it is specific to D-Wave, the point of QUBOTools is that people could also use this to transform any input into any output. Hence if someone likes CIM, but wants to use DWave, they could just go to QUBOTools for that transformation.
I think that it would be in place to open an issue on MiniZinc.jl and have them deal with this. I agree that the effort of maintaining reading working is a major task, so let's keep what there is there and have nice error messages pointing to other places (ahem MiniZing.jl) to write their programs.